lib/exit.fz

```diff
@@ -86,3 +86,49 @@ exit_type is
     #
     exit(code i32) =>
       fuzion.std.exit code
+
+
+# delayed_exit -- effect that runs some code that can set an exit code, but
+# only exits after all the code has been run
+#
+# this is intended for use in tests
+#
+delayed_exit (_ unit) : effect effectMode.plain is
+
+
+  # mutable value of the current exit code set
+  #
+  exit_code := mut 0
+
+
+  # update the exit code to the given value
+  #
+  set0(code i32) =>
```

Comment:
I think the common case and what probably most would expect is that the first none zero exit code is the one that is latter used at exit?